include/operations/blas2_trees.h

```diff
@@ -227,6 +227,45 @@ make_Gemv_Row(lhs_t &lhs_, matrix_t &matrix_, vector_t &vector_,
                            local_memory_size_);
 }
 
+/**
+ * @struct Gbmv
+ * @brief Tree node representing a band matrix_ vector_ multiplication.
+ */
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_transposed>
+struct Gbmv {
+  using value_t = typename vector_t::value_t;
+  using index_t = typename vector_t::index_t;
+
+  lhs_t lhs_;
+  matrix_t matrix_;
+  index_t kl_;
+  index_t ku_;
+  vector_t vector_;
+
+  Gbmv(lhs_t &_l, matrix_t &_matrix, index_t &_kl, index_t &_ku,
+       vector_t &_vector);
+  index_t get_size() const;
+  bool valid_thread(cl::sycl::nd_item<1> ndItem) const;
+  value_t eval(index_t i);
+  value_t eval(cl::sycl::nd_item<1> ndItem);
+  template <typename sharedT>
+  value_t eval(sharedT shrMem, cl::sycl::nd_item<1> ndItem);
+  void bind(cl::sycl::handler &h);
+  void adjust_access_displacement();
+};
+/*!
+ @brief Generator/factory for GBMV trees.
+ */
+template <uint32_t local_range, bool is_transposed, typename lhs_t,
+          typename matrix_t, typename vector_t>
+Gbmv<lhs_t, matrix_t, vector_t, local_range, is_transposed> make_Gbmv(
```

Comment:
Why do you need make_Gbmv? you don't need that function as this function returns gbmv. Also, this GBMV class called internally so no need for this function

Reply:
I think it makes sense to keep this helper function. The advantage is that it can automatically deduce the template parameters `lhs_t`, `matrix_t` and `vector_t`. It's used quite a lot for `makeGemv` for instance but you're right that it's used internally only.

Reply:
👍